central/sensor/service/service_impl.go

```diff
@@ -116,10 +116,16 @@ func (s *serviceImpl) Communicate(server central.SensorService_CommunicateServer
 	// Note: New clusters which have never been properly connected to central have their HealthStatus.LastContact empty.
 	// If, for whatever reason, a cluster needs to re-retrieve CRS-issued service certificates, the cluster can do so
 	// as long as that cluster has never successfully connected to central with proper service certificates
-	// (not ServiceType_REGISTRANT_SERVICE).
-	if svcType == storage.ServiceType_REGISTRANT_SERVICE && cluster.GetHealthStatus().GetLastContact() != nil {
-		log.Errorf("It is forbidden to connect with a Cluster Registration Certificate as already-existing cluster %q.", cluster.GetName())
-		return errox.NotAuthorized.CausedByf("forbidden to use a Cluster Registration Certificate for already-existing cluster %q", cluster.GetName())
+	// (not ServiceType_REGISTRANT_SERVICE) and the deployment identification is the same.
+	if svcType == storage.ServiceType_REGISTRANT_SERVICE {
+		if cluster.GetHealthStatus().GetLastContact() != nil {
+			log.Errorf("It is forbidden to connect with a Cluster Registration Certificate as already-existing cluster %q.", cluster.GetName())
+			return errox.NotAuthorized.CausedByf("forbidden to use a Cluster Registration Certificate for already-existing cluster %q", cluster.GetName())
+		}
+		if !cluster.GetMostRecentSensorId().EqualVT(sensorHello.GetDeploymentIdentification()) {
```

Comment:
This, among other things, checks for a matching k8s node name. AFAICT this means that if the sensor pod gets rescheduled between registration and persisting its certificates, this change would prevent registration completion. I think we don't want to be *that* strict 🤔 

Similarly, the `app_*` fields might change if the user decided to "fix things" by deleting and re-installing with the same parameters. This I think is borderline case...

Perhaps we should be checking `{system,default}_namespace_id` as a poor-man's check of cluster identity, but I'm not sure whether in the long run it would bring more harm than good..

Maybe instead we should have a documented time limit - registration needs to complete within x minutes/hours from cluster creation?